lib/linter.js

```diff
@@ -83,14 +84,45 @@ exports.lint = function (input, fullPath, config) {
     Object.freeze(ast);
 
     ast.traverse(function (node) {
-        var i;
-
-        for (i = 0; i < linters.length; i++) {
-            results.push(linters[i].call(null, {
-                config: config,
-                node: node
-            }));
-        }
+        linters.forEach(function (linter) {
+            var lint;
+
+            // Bail if the linter isn't wanted
+            if (!config[linter.name] || (config[linter.name] && !config[linter.name].enabled)) {
+                return;
+            }
+
+            if (linter.nodeTypes && Array.isArray(linter.nodeTypes) && linter.nodeTypes.length) {
```

Comment:
I think we could remove the `length` check, `indexOf` will return `-1` on an empty array anyway,